### ISSUE TITLE
[BUG] npm does not create --prefix dir

### ISSUE THREAD

**Issue by cb1kenobi** [npm/cli#2012]
### Current Behavior:
npm 7.0.3 errors because `--prefix` directory does not exist.

This is appears to be a regression as it does not happen with npm 6 and I was unable to find a ticket or mention in the npm 7 release notes that `--prefix` no longer creates the destination directory.

```
$ npm install <ANY_PACKAGE_NAME> --no-audit --no-save --production --prefix /Users/joe/foo
```

```
npm install exited (code 254)
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /Users/joe/foo
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/Users/joe/foo'
npm ERR! enoent This is related to npm not being able to find a file.
```

If I manually create the prefix directory, then run `npm install`, it works fine.

### Expected Behavior:
The package should install into the directory specified by the `--prefix` argument and the output should look similar to:

```
npm install exited (code 0)
<PACKAGE_NAME>@<VERSION> install /Users/joe/foo
```

### Steps To Reproduce:
1. Run `npm install <some_package> --prefix /some/path/that/does/not/exist`
2. See error

### Environment:
- OS: macOS 10.15.7 (catalina)
- Node: 14.13.0
- npm: 7.0.3

**Comment by viceice** [comment]
Same here:

```Dockerfile
FROM node:15
ENV NPM_CONFIG_PREFIX=/tmp/.npm
RUN npm install -g yarn
```

```
#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 107B done
#2 DONE 0.0s

#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#3 [internal] load metadata for docker.io/library/node:15
#3 DONE 1.2s

#5 [1/2] FROM docker.io/library/node:15@sha256:9a7682432998d34759cf0b0d079e...
#5 CACHED

#4 [2/2] RUN npm install -g yarn
#4 0.924 npm notice
#4 0.924 npm notice New patch version of npm available! 7.0.3 -> 7.0.5
#4 0.924 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.0.5>
#4 0.925 npm notice Run `npm install -g npm@7.0.5` to update!
#4 0.925 npm notice
#4 0.925 npm ERR! code ENOENT
#4 0.925 npm ERR! syscall lstat
#4 0.925 npm ERR! path /tmp/.npm
#4 0.926 npm ERR! errno -2
#4 0.926 npm ERR! enoent ENOENT: no such file or directory, lstat '/tmp/.npm'
#4 0.927 npm ERR! enoent This is related to npm not being able to find a file.
#4 0.927 npm ERR! enoent
#4 0.936
#4 0.936 npm ERR! A complete log of this run can be found in:
#4 0.936 npm ERR!     /root/.npm/_logs/2020-10-24T08_39_41_479Z-debug.log
#4 ERROR: executor failed running [/bin/sh -c npm install -g yarn]: runc did not terminate sucessfully
------
 > [2/2] RUN npm install -g yarn:
------
failed to solve with frontend dockerfile.v0: failed to build LLB: executor failed running [/bin/sh -c npm install -g yarn]: runc did not terminate sucessfully
```

**Comment by ruyadorno** [comment]
@cb1kenobi Thanks for the report! This should have been fixed in `npm@7.4.0` 😊
